packages/ingest/parse-pdr.js

```diff
@@ -110,61 +120,63 @@ module.exports.parsePdr = async function parsePdr(pdrFilePath, collection, pdrNa
     throw new PDRParsingError(e.message);
   }
 
-  // check if the PDR has groups
-  // if so, get the objects inside the first group
-  // TODO: handle cases where there are more than one group
-  const groups = parsed.groups();
-  if (groups.length > 0) {
-    parsed = groups[0];
-  }
+  return parsed;
+}
 
-  // Get all the file groups
-  const fileGroups = parsed.objects('FILE_GROUP');
+function granuleFromFileGroup(fileGroup, granuleIdExtraction, pdrName) {
+  if (!fileGroup.get('DATA_TYPE')) throw new PDRParsingError('DATA_TYPE is missing');
+  const dataType = fileGroup.get('DATA_TYPE').value;
 
-  for (const group of fileGroups) {
-    // get all the file specs in each group
-    const specs = group.objects('FILE_SPEC');
-    let dataType = group.get('DATA_TYPE');
+  // get all the file specs in each group
+  const specs = fileGroup.objects('FILE_SPEC');
+  // FIXME This is a very generic error
+  if (specs.length === 0) throw new Error();
 
-    if (!dataType) throw new PDRParsingError('DATA_TYPE is missing');
+  const files = specs.map(parseSpec.bind(null, pdrName));
+  const granuleId = extractGranuleId(files[0].name, granuleIdExtraction);
```

Comment:
Is it possible files would be empty here? Is that something we should check for?